app/app.h

```diff
@@ -4,5 +4,52 @@
 
 namespace vsite::oop::v4 
 {
+    class animal {
+    public:
+        virtual ~animal() = default;
+        virtual std::string species() const = 0;
+        virtual unsigned legs() const = 0;
+    };
 
+    class insect : public animal {
+    public:
+        virtual ~insect() = default;
+    };
+
+    class spider : public animal {
+    public:
+        virtual ~spider() = default;
+    };
+
+    class bird : public animal {
+    public:
+        virtual ~bird() = default;
+    };
+
+    class cockroach : public insect {
+    public:
+        std::string species() const override;
+        unsigned legs() const override;
```

Comment:
Ideja klasa poput `insect`/`spider`/`bird` je da tamo implementirate ono što im je zajedničko - broj nogu (da se ne mora za svaku konkretnu pticu ili pauka ponovo).